commit 744735550d4a4fd6d4be40776069d799dca5ee39
Author: Ulrich Weigand <uweigand@de.ibm.com>
Date:   Mon Jun 14 16:09:55 2010 +0000

    	* cp-valprint.c (cp_print_static_field): Members of
    	dont_print_stat_array_obstack are of type "struct type *".
    	(_initialize_cp_valprint): Likewise.

### a/gdb/ChangeLog
### b/gdb/ChangeLog
## -1,5 +1,11 @@
 2010-06-14  Ulrich Weigand  <uweigand@de.ibm.com>
 
+	* cp-valprint.c (cp_print_static_field): Members of
+	dont_print_stat_array_obstack are of type "struct type *".
+	(_initialize_cp_valprint): Likewise.
+
+2010-06-14  Ulrich Weigand  <uweigand@de.ibm.com>
+
 	* frame.c (frame_register_unwind): Do not access contents
 	of "optimized out" unwound register value.
 
--- a/gdb/cp-valprint.c
+++ b/gdb/cp-valprint.c
@@ -615,7 +615,7 @@ cp_print_static_field (struct type *type,
       first_dont_print
 	= (struct type **) obstack_base (&dont_print_stat_array_obstack);
       i = obstack_object_size (&dont_print_stat_array_obstack)
-	/ sizeof (CORE_ADDR);
+	/ sizeof (struct type *);
 
       while (--i >= 0)
 	{
@@ -764,7 +764,7 @@ Show printing of object's derived type based on vtable info."), NULL,
 			   show_objectprint,
 			   &setprintlist, &showprintlist);
 
-  obstack_begin (&dont_print_stat_array_obstack, 32 * sizeof (CORE_ADDR));
+  obstack_begin (&dont_print_stat_array_obstack, 32 * sizeof (struct type *));
   obstack_begin (&dont_print_statmem_obstack, 32 * sizeof (CORE_ADDR));
   obstack_begin (&dont_print_vb_obstack, 32 * sizeof (struct type *));
 }
